contrib/devtools/lint-includes.sh

```diff
@@ -0,0 +1,39 @@
+#!/bin/bash
+#
+# Copyright (c) 2017 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+#
+# Check for duplicate includes.
+
+EXIT_CODE=0
+for HEADER_FILE in $(git ls-files | grep -E "^src/.*\.h$" | grep -Ev "/(leveldb|secp256k1|univalue)/"); do
```

Comment:
nit: you could do this and the change below as a function, and pass in the .h or .cpp'ness in a bash function, e.g.

```bash
filter_suffix() { git ls-files | grep -E "^src/.*\.${1}"'$' | grep -Ev "/(leveldb|secp256k1|univalue)/"; }
```

And then below would become:
```bash
for HEADER_FILE in $(filter_suffix h); do
...

for CPP_FILE in $(filter_suffix cpp); do
...
```